logstash-core-event-java/src/main/java/com/logstash/Event.java

```diff
@@ -167,10 +170,35 @@ public boolean includes(String reference) {
         }
     }
 
+    public byte[] toBinary() throws IOException {
+        HashMap<String, Map<String, Object>> hashMap = new HashMap<>();
+        hashMap.put("DATA", this.data);
+        hashMap.put("META", this.metadata);
+        return CBOR_MAPPER.writeValueAsBytes(hashMap);
+    }
+
+    public static Event fromBinary(byte[] source) throws IOException {
+        if (source == null || source.length == 0) {
+            return new Event();
+        }
+        Object o = CBOR_MAPPER.readValue(source, HashMap.class);
+        if (o == null) {
+            throw new IOException("The binary source deserialized to null");
+        }
+        try {
+            HashMap<String, Map<String, Object>> hashMap = (HashMap<String, Map<String, Object>>) o;
```

Comment:
do we have to be implementation specific here? can we just cast it to any `Map`?